src/spanned.rs

```diff
@@ -130,7 +142,8 @@ where
                 span = joined;
             }
         }
-        span
+
+        check_invalid_spans(span)
```

Comment:
This check should happen above in the loop so that a syntax tree node can still have a valid span even if there are fragments interpolated into it that do not have valid spans. Let's skip any invalid spans when doing the joins.